python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -121,6 +128,35 @@ def _exec_sling_cmd(
             if proc.returncode != 0:
                 raise Exception("Sling command failed with error code %s", proc.returncode)
 
+    def replicate(
+        self,
+        *,
+        dagster_sling_translator: DagsterSlingTranslator,
+        source_stream: str,
+        target_object: str,
+        mode: SlingMode,
+        primary_key: Optional[List[str]] = None,
+        update_key: Optional[str] = None,
+        source_options: Optional[Dict[str, Any]] = None,
+        target_options: Optional[Dict[str, Any]] = None,
+    ):
+        logs = ""
+        for line in self.sync(
+            source_stream,
+            target_object,
+            mode,
+            primary_key,
+            update_key,
+            source_options,
+            target_options,
+        ):
+            logger.info(line)
```

Comment:
Thoughts on piping this to `stdout` by default, instead of the event log? Is it necessary to capture all of these lines and add them as metadata to the `MaterializeResult`?